aten/src/ATen/native/DispatchStub.h

```diff
@@ -268,6 +284,9 @@ struct RegisterHIPDispatch {
 // is HIP in the PyTorch HIPify build.
 #define REGISTER_DISPATCH(name, fn) REGISTER_CUDA_DISPATCH(name, fn)
 // #define REGISTER_DISPATCH(name, fn) REGISTER_HIP_DISPATCH(name, fn)
+#elif defined(__OBJC__)
```

Comment:
We should use `USE_MPS` here to make it specific to MPS backend.

Reply:
Wouldn't `USE_MPS` overwrite all other REGISTER_DISPATCH calls that the cpu makes? If `USE_MPS` is always defined (even for the cpu cpp files importing this DispatchStub.h), then it would fail

Reply:
Thanks for pointing this out - I've added this guard macro (USE_MPS_ to the MPS code from DispatchStub